dc-model/src/main/java/de/digitalcollections/model/api/filter/FilterCriteria.java

```diff
@@ -0,0 +1,47 @@
+package de.digitalcollections.model.api.filter;
+
+import de.digitalcollections.model.api.filter.enums.FilterOperation;
+import java.util.Collection;
+
+/**
+ * Filter Criteria Container. A filter criteria is a composition of
+ *
+ * <ul>
+ *   <li>a field name being target of filter operation
+ *   <li>a filter operation
+ *   <li>one ore more filter values to be used for filtering operation
+ * </ul>
+ *
+ * @param <T> type of field-data to be filtered (operand), e.g. Date, LocalDate, Long, etc.
+ */
+public interface FilterCriteria<T extends Comparable> {
```

Comment:
- Wäre das vom Namen her nicht ein `FilterCriterion`? Ist ja nur ein Kriterium, also Singular.
- Warum muss `T` ein `Comparable` sein? Das Modell unterstützt ja auch `eq` und andere Operationen die kein `Comparable` voraussetzen.
- Wie stellst du dir vor dass z.B. der `IN` Operator so umgesetzt werden kann? Das bräuchte ja einen Container-Typen, und ein `Set` oder auch nur ein `String[]` unterstützen ja kein `Comparable`

Reply:
- Name auf FilterCriterion geändert
- kick ass generics as far as possible (T nur noch als Typangabe für Spring MVC conversion service benötigt): nur noch Object oder Comparable oder beliebige Collection
- IN kann nun mit beliebiger Collection umgehen
- Interface weggeschmissen, nur noch Klasse
- setter-methoden (bis auf dieldName) entfernt, jackson benutzt jetzt auch constructor
- Validierung (palusi-prüfung) basierend auf neuem operandCount Enum umgesetzt. Es wird IllegalArgumentException geschmissen, wenn im Konstruktor übergebene argumente nicht zum operation-typ passen

done. let's merge :-)